### PR TITLE
Dynamic concept authorities

### PR DESCRIPTION
**Pull request by MiroslavGatsanoga**
# Description

## What

Migrate concept authorities to ontology defined config.

## Why

[JIRA ticket](https://financialtimes.atlassian.net/browse/UPPSF-2467)

## Anything, in particular, you'd like to highlight to reviewers

Had to fix some tests along the way, please let me know if I missed to move some authority related code to the ontology.

## Scope and particulars of this PR (Please tick all that apply)

- [ ] Tech hygiene (dependency updating & other tech debt)
- [ ] Bug fix
- [X] Feature
- [ ] Documentation
- [ ] Breaking change
- [ ] Minor change (e.g. fixing a typo, adding config)

___
This Pull Request follows the rules described in our [Pull Requests Guide](https://github.com/Financial-Times/upp-docs/tree/master/guides/pr-guide)

### CODE REVIEW

File: concepts/concepts_service.go [hunk 1]
```diff
@@ -407,28 +407,30 @@ func validateObject(aggConcept ontology.NewAggregatedConcept, transID string) er
...
-		if concept.AuthorityValue == "" {
-			return requestError{formatError("sourceRepresentation.authorityValue", concept.UUID, transID)}
+
+		if sourceConcept.Type == "" {
+			return requestError{formatError("sourceRepresentation.type", sourceConcept.UUID, transID)}
 		}
-		if _, ok := constraintMap[concept.Type]; !ok {
+
+		if _, ok := constraintMap[sourceConcept.Type]; !ok {
 			return requestError{formatError("type", aggConcept.PrefUUID, transID)}
```
Comment (ivanruski) [inline concepts/concepts_service.go:426]: **suggestion(non-blocking):** Currently(If I am understanding the code correctly), if we try to send a payload with `type: ""` for the `sourceConcept` we would get: `"Invalid request, no sourceRepresentation.type has been supplied"`, but if we pass an invalid `type` e.g. `BrandZ`, we would get `"Invalid request, no type has been supplied"`.
We could return a more accurate error message.

File: concepts/concepts_service_test.go [hunk 2]
```diff
@@ -1968,149 +1965,151 @@ func TestTransferCanonicalMultipleConcordance(t *testing.T) {
...
+		t.Run(test.name, func(t *testing.T) {
+			newAggConcept := ontology.TransformToNewAggregateConcept(test.aggConcept)
+			err := validateObject(newAggConcept, "transaction_id")
+			if err != nil {
+				assert.NotEmpty(t, test.returnedError, "test.returnedError should not be empty when there is an error")
+				assert.Contains(t, err.Error(), test.returnedError, test.name)
+			} else {
+				assert.Empty(t, test.returnedError, "test.returnedError should be empty when there is no error")
+				assert.NoError(t, err, test.name)
+			}
```
Comment (ivanruski) [inline concepts/concepts_service_test.go:2111]: **nitpick(non-blocking):** In my opinion `test.returnedError` should drive whether or not the test is correct. In other words, if `test.returnedError != ""`, then `err` must not be `nil`

**Review by ivanruski — APPROVED** [review]

File: concepts/concepts_service_test.go (hunk 2 above)
Comment (ivan-p-nikolov) [reply concepts/concepts_service_test.go:2111]: What's the difference?

**Review by ivan-p-nikolov — APPROVED** [review]
LGTM

File: concepts/concepts_service.go (hunk 1 above)
Comment (MiroslavGatsanoga) [reply concepts/concepts_service.go:426]: Yes, I noticed that but decided to leave it for the task that will be dealing with the type of the concepts.

File: concepts/concepts_service_test.go (hunk 2 above)
Comment (MiroslavGatsanoga) [reply concepts/concepts_service_test.go:2111]: I don't think there is a difference, just a preference as I see it, I will leave it as it is for now.

**Pull request merged**